core/amber/src/main/scala/edu/uci/ics/amber/engine/common/promise/PromiseInvocation.scala

```diff
@@ -0,0 +1,7 @@
+package edu.uci.ics.amber.engine.common.promise
+
+/** The invocation of a promise
+  * @param context
+  * @param call
+  */
+case class PromiseInvocation(context: PromiseContext, call: PromiseBody[_]) extends PromisePayload
```

Comment:
This is a ControlInvocation. Also I dont think this will be needed if we name control payload as (ctx, command) because this is essentially the same.

Reply:
then what should we do for those ReturnPayloads?